### ISSUE TITLE
Hello, there is a problem with self.config.use_fast_mode=True in the switch-transformers code you gave.

### ISSUE THREAD

**Issue by CloudComputing-linux** [Bumble666/Hyper_MoE#4]
class ParameterGenerator(nn.Module):
    def __init__(self, config, input_size, output_size):
        # output_dim: Main Network output_dim
        # input_dim: Main Network in_dim
        super().__init__()
        self.config = config
        self.layer_embed = nn.Embedding(config.num_hidden_layers, config.layer_emb_dim)
        self.decoder = SimpleGenerator(
            config, input_size, output_size
        )

    def forward(self, hidden_inputs, layer_idx):
        """

        :param hidden_inputs: [batch_size，seq_len,experts_embedding_dim]
        :param layer_idx:
        :return:
        """
        if self.config.use_fast_mode:
            layer_idx = torch.ones(self.config.num_experts, dtype=torch.long,
                                device=hidden_inputs.device) * layer_idx
 
        else:
            layer_idx = torch.ones(hidden_inputs.size(0), hidden_inputs.size(1), dtype=torch.long, device=hidden_inputs.device) * layer_idx
        layer_inputs = self.layer_embed(layer_idx) # shape [num_experts,layer_emb_dim]
        hidden_inputs = torch.cat([hidden_inputs, layer_inputs], dim=-1)   ##There is no guarantee of shape consistency here
        out = self.decoder(hidden_inputs)
        return out
